### PR TITLE
Add support for memo

### PR DESCRIPTION
**Pull request by madcake**

### CODE REVIEW

File: src/proto/Stellar.proto [hunk 1]
```diff
@@ -15,6 +15,10 @@ message SigningInput {
 
     string destination = 5;
 
+    int32 memo_type = 6;
```
Comment (vikmeup) [inline src/proto/Stellar.proto]: I think you can create protobuf enum?

File: src/Stellar/Signer.cpp [hunk 2]
```diff
@@ -63,3 +64,35 @@ void Signer::encodeAddress(Address address, Data& data) const {
...
+        case TWStellarMemoTypeText: {
+            encode32BE(static_cast<uint32_t>(memoData.size()) - 1, data);
+            data.insert(data.end(), memoData.begin(), memoData.end() - 1);
+        } break;
+        case TWStellarMemoTypeHash:
+        case TWStellarMemoTypeReturn:
+            data.insert(data.end(), memoData.begin(), memoData.end());
+            break;
+        case TWStellarMemoTypeId:
+            data.insert(data.end(), memoData.begin(), memoData.end());
```
Comment (hewigovens) [inline src/Stellar/Signer.cpp]: Same as hash/return?

File: src/Stellar/Signer.cpp [hunk 3]
```diff
@@ -63,3 +64,35 @@ void Signer::encodeAddress(Address address, Data& data) const {
...
+    pad(data);
+}
+
+void Signer::pad(Data& data) const {
+    int pad = 0;
+    int mod = static_cast<int>(data.size() % 4);
+    if (mod > 0) {
+        pad = 4 - mod;
+    }
+    while (pad-- > 0) {
```
Comment (hewigovens) [inline src/Stellar/Signer.cpp]: pad —

File: src/Stellar/Signer.cpp (hunk 2 above)
Comment (madcake) [reply src/Stellar/Signer.cpp]: Yes it's same only type different

**Pull request merged**